src/main/java/io/openmanufacturing/ame/repository/strategy/LocalFolderResolverStrategy.java

```diff
@@ -151,6 +162,20 @@ public LocalPackageInfo getLocalPackageInformation( final String storagePath ) {
             getNonTurtleFiles( storagePath, file ) );
    }
 
+   @Override
+   public AspectModelUrn convertFileToUrn( final File inputFile ) {
+      final File versionDirectory = inputFile.getParentFile();
+      final String version = versionDirectory.getName();
+      final File namespaceDirectory = versionDirectory.getParentFile();
+      final String namespace = namespaceDirectory.getName();
+      final String aspectName = FilenameUtils.removeExtension( inputFile.getName() );
+      final String urn = String.format( "urn:bamm:%s:%s#%s", namespace, version, aspectName );
+      return new SdsAspectMetaModelResourceResolver().getAspectModelUrn( urn ).getOrElse( () -> {
```

Comment:
For the user it is useful to get an exact error message in the frontend, so here the method: new SdsAspectMetaModelResourceResolver().getAspectModelUrn( urn ) is very clever, because here we can very easily check if the urn can be created and if not throw an own exception.